src/Apps/Sell/ArtworkFormContext.tsx

```diff
@@ -0,0 +1,109 @@
+import { UpdateSubmissionMutationInput } from "__generated__/UpdateConsignSubmissionMutation.graphql";
+import { createOrUpdateConsignSubmission } from "Apps/Consign/Routes/SubmissionFlow/Utils/createOrUpdateConsignSubmission";
+import { createContext, useContext, useEffect, useState } from "react";
+import { useRouter } from "System/Router/useRouter";
+import { useSystemContext } from "System/SystemContext";
+
+interface Actions {
+  goToPreviousStep: () => void
+  goToNextStep: () => void
+  updateSubmission: (values: UpdateSubmissionMutationInput) => void
+}
+
+interface State {
+  atFirstStep: boolean
+  atLastStep: boolean
+  currentStep: string
+  submissionID: string | undefined
+}
+interface ArtworkFormContextProps {
+  actions: Actions
+  state: State
+}
+
+export const ArtworkFormContext = createContext<ArtworkFormContextProps>({} as any)
+
+export const STEPS = [
+  "title",
+  "photos",
+  "details",
+  "purchase-history",
+  "dimensions"
+]
+
+interface ArtworkFormContextProviderProps {
+  children: React.ReactNode
+  submissionID?: string
+}
+
+export const ArtworkFormContextProvider: React.FC<ArtworkFormContextProviderProps> = ({ children, submissionID }) => {
+  const { match, router } = useRouter()
+  const { relayEnvironment } = useSystemContext()
+  const [currentStep, setCurrentStep] = useState<string>(match.location.pathname.split("/").pop() ?? "")
+  const [atFirstStep, setAtFirstStep] = useState<boolean>(false)
+  const [atLastStep, setAtLastStep] = useState<boolean>(false)
```

Comment:
Should just use a single state for all the steps that's a union.

Or we use `[useCursor](https://github.com/artsy/force/blob/dfb2f42b0a6c3a51d1310b457660202a3d694961/src/Components/HeroCarousel/HeroCarouselLarge.tsx#L42-L47)` for cycling through an index

Reply:
Ahh `useCursor` is very cool. Yeah, I'm aware that this initial state tracking is rough. I anticipate we'll improve upon this as we iterate.

Do you feel strongly about seeing a different state tracking implementation before merge?

Reply:
I mean: the way this state is implemented can absolutely lead to invalid states and is just confusing. There's never a reason to have so many `useState` declarations like that.

Reply:
I took a pass at converting this over to utilize useCursor. Let me know if anything else jumps out at you.